bundles/io/org.eclipse.smarthome.io.transport.serial/src/main/java/org/eclipse/smarthome/io/transport/serial/SerialPortIdentifier.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Interface of a serial port identifier.
+ *
+ * @author Markus Rathgeb - Initial contribution
+ */
+@NonNullByDefault
+public interface SerialPortIdentifier {
+    /**
+     * Gets the name of the port.
+     *
+     * @return the port's name
+     */
+    String getName();
+
+    /**
+     * Opens a serial port for communicating.
+     *
+     * @param owner name of the owner that port should be assigned to
+     * @param timeout time in milliseconds to block waiting for opening the port
+     * @return a serial port
```

Comment:
@throws NPE or annotations on `owner`

Reply:
Have you realized that the whole class is annotated by `NonNullByDefault`? No need to add the `NonNull` annotation to any members, only `Nullable` would be needed.

Reply:
Didn't notice at all. Thanks for the clarification.